tt_metal/api/device.hpp

```diff
@@ -0,0 +1,342 @@
+ *
+ * @return Size_t representing the number of PCIe devices.
+ */
+size_t GetNumPCIeDevices();
+
+/**
+ * @brief Retrieves the PCIe device ID for a given device ID.
+ *
+ * @param device_id ID of the device to query.
+ * @return Chip ID of the PCIe device.
+ */
+chip_id_t GetPCIeDeviceID(chip_id_t device_id);
+
+/**
+ * @brief Instantiates a Device object.
+ *
+ * @param device_id ID of the device to target (0 to GetNumAvailableDevices() - 1).
+ * @param num_hw_cqs Number of hardware command queues (default: 1, valid range: 1 to 2).
+ * @param l1_small_size L1 small space to reserve (default: DEFAULT_L1_SMALL_SIZE).
+ * @param trace_region_size Trace region size to reserve (default: DEFAULT_TRACE_REGION_SIZE).
+ * @param dispatch_core_type Dispatch core type to use (default: DispatchCoreType::WORKER).
+ * @return Device handle to the created device.
+ */
+Device CreateDevice(
+    chip_id_t device_id,
+    uint8_t num_hw_cqs = 1,
+    size_t l1_small_size = DEFAULT_L1_SMALL_SIZE,
+    size_t trace_region_size = DEFAULT_TRACE_REGION_SIZE,
+    DispatchCoreType dispatch_core_type = DispatchCoreType::WORKER);
+
+/**
+ * @brief Resets and closes the device.
+ *
+ * @param device Handle to the device to close.
+ * @return True if the device was successfully closed; otherwise, false.
+ */
+bool CloseDevice(Device device);
+
+
+/**
+ * @brief Launches a program on the device.
+ *
+ * @param device The device on which to launch the program.
+ * @param program The program to execute.
+ * @param wait_until_cores_done If true, waits until cores have completed execution.
+ * @param force_slow_dispatch If true, forces slow dispatch mode.
+ */
+void LaunchProgram(Device device, Program program, bool wait_until_cores_done = true, bool force_slow_dispatch = false);
+
+/**
+ * @brief Dumps device-side profiler data to a CSV log.
+ *
+ * @param device The device holding the program being profiled.
+ * @param worker_cores CoreRangeSet of worker cores being profiled.
+ * @param last_dump If true, indicates the last dump before process termination.
+ */
+void DumpDeviceProfileResults(Device device, const CoreRangeSet &worker_cores, bool last_dump = false);
+
+
+/**
+ * @brief Retrieves the architecture of the device.
+ *
+ * @param device The device to query.
+ * @return ARCH representing the device architecture.
+ */
+ARCH GetArch(Device device);
+
+/**
+ * @brief Retrieves the ID of the device.
+ *
+ * @param device The device to query.
+ * @return Chip ID of the device.
+ */
+chip_id_t GetId(Device device);
+
+/**
+ * @brief Retrieves the number of DRAM channels on the device.
+ *
+ * @param device The device to query.
+ * @return Number of DRAM channels.
+ */
+int GetNumDramChannels(Device device);
+
+/**
+ * @brief Retrieves the available L1 size per worker core on the device.
+ *
+ * @param device The device to query.
+ * @return L1 size per core in bytes.
+ */
+uint32_t GetL1SizePerCore(Device device);
```

Comment:
about here I'm wondering why we have so many comments and why this is spread out so much.  what if we had a comment block that essentially said "getters" and then 1 getter on each line w/ a small comment if needed - the names are pretty self explanatory and the comments mostly redundant.

packing them would make the code more discoverable I think

Reply:
The comments show up in static analysis when you hover over callsites, whereas if you group them without comments, you don't get a nice annotation in your static analysis tooltips